ggshield/cmd/utils/utils.py

```diff
@@ -0,0 +1,14 @@
+import re
+from pathlib import Path
+from typing import Set, Union
+
+from click import UsageError
+
+from ggshield.utils.files import is_filepath_excluded
+
+
+def assert_directory_not_ignored(
+    directory: Union[str, Path], exclusion_regexes: Set[re.Pattern]
+):
+    if is_filepath_excluded(directory, exclusion_regexes):
+        raise UsageError("An ignored directory cannot be scanned.")
```

Comment:
Is it always a directory ?
Cannot it be a file sometime ?

Reply:
It can be a file in the secrets path scan. I will change it to "file or directory".